experimenter/experimenter/experiments/changelog_utils.py

```diff
@@ -87,3 +91,60 @@ def generate_nimbus_changelog(experiment, changed_by, message, changed_on=None):
         message=message,
         changed_on=changed_on,
     )
+
+
+def get_formatted_change_object(field_name, field_diff, changelog, timestamp):
+    event = NimbusConstants.ChangeEvent.find_enum_by_key(field_name.upper())
+    field_display_name = (
+        event.display_name
+        if event.value != NimbusConstants.ChangeEvent.TRIVIAL.value
+        else field_name
+    )
```

Comment:
Instead of this, we can find field information via:

```python
f = NimbusExperiment._meta.get_field(field_name)
```

Then you can check for the field types you're looking for, e.g., 

```python
if isinstance(f, django.db.models.fields.BooleanField):
    pass
elif isinstance(f, django.db.models.fields.related.ManyToManyField): # list
    pass
elif isinstance(f, django.db.models.fields.json.JSONField): # JSON
    pass
```

For the field descriptions, you can add [`label`](https://docs.djangoproject.com/en/4.2/ref/forms/fields/#label) to each field and use that for the field name if present.

Reply:
Oh, I tried using the `isinstance` method earlier but it didn't work as I was passing the field directly instead of its instance. Still we do need to group them by events so that they can be rendered appropriately in the UI. Should I create a Enum which categorises each field type with an event then ?

Reply:
Sorry I sent a link to the wrong docs. You can use the `name` or [`verbose_name`](https://docs.djangoproject.com/en/4.2/ref/models/fields/#verbose-name) attributes. [More docs](https://docs.djangoproject.com/en/4.2/topics/db/models/#verbose-field-names)